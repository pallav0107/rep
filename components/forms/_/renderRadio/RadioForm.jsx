import React from 'react';
import { Radio } from '@wdpui/react-gel';
import PropTypes from 'prop-types';
import { IconAlert } from '@wdpui/react-gel';
import styled from 'styled-components';
import { Box } from 'grid-styled';

const StyledRadio = styled(Radio)`
  word-wrap: break-word;
`;
export const ErrorIcon = styled(IconAlert)`
  color: ${({ theme }) => theme.color.danger};
  margin-bottom: 3px;
`;

const RadioForm = ({ input, data ,flip,alertType}) => (
  <Box>
    {data.map(key => (
      <StyledRadio {...input} label={key} value={key} flip={flip} alertType={alertType} mx={4} mr={5} />
    ))}
  </Box>
);

RadioForm.propTypes = {
  label: PropTypes.string,
  data: PropTypes.shape({}).isRequired,
};

RadioForm.defaultProps = {
  label: '',
  name: '',
  value: '',
};

export default RadioForm;
