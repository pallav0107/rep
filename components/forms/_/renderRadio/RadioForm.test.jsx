import React from 'react';
import { shallow } from 'enzyme';
import RadioForm from './RadioForm';
import { withTheme } from '../../../../../utils/jest/TestUtils';
import "../../../../setupTests"

describe('RadioForm', () => {
  it('should render the RadioForm', () => {
    const component = withTheme(<RadioForm />);
    const wrapper = shallow(component);
    expect(wrapper).toMatchSnapshot();
  });
});
