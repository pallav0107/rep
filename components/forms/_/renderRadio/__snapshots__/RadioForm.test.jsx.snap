// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioForm should render the RadioForm 1`] = `
<RadioForm
  label=""
  name=""
  value=""
/>
`;
